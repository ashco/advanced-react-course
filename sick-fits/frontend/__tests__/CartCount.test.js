import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import CartCount from '../components/CartCount';

describe('<CartCount>', () => {
  // always start with just checking if it renders fine
  it('renders', () => {
    // Dont even need an expect because if something goes wrong with below then it will throw error and test will fail
    shallow(<CartCount count={10} />);
  });

  it('matches the snapshot', () => {
    const wrapper = shallow(<CartCount count={11} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  // SHALLOW - only goes 1 layer deep
  // it('updates via props', () => {
  //   const wrapper = shallow(<CartCount count={50} />);
  //   expect(toJSON(wrapper)).toMatchSnapshot();
  //   wrapper.setProps({ count: 10 });
  //   expect(toJSON(wrapper)).toMatchSnapshot();
  // });

  // MOUNT - will run it in browser environment - 'tests should be as similar to what the user will be experiencing, so mount! - Wes Bos'
  it('updates via props', () => {
    const wrapper = mount(<CartCount count={50} />);
    // console.log(wrapper.debug());
    return;
    expect(toJSON(wrapper)).toMatchSnapshot();
    // wrapper.setProps({ count: 10 });
    // expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
