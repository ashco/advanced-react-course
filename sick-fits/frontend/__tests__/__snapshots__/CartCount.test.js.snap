// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount> matches the snapshot 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="11"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        11
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;
